src/commands/ext-install.ts

```diff
@@ -114,17 +116,58 @@ async function installExtension(options: InstallExtensionOptions): Promise<void>
 /**
  * Command for installing a extension
  */
-export default new Command("ext:install <extensionName>")
-  .description("install an extension, given <extensionName> or <extensionName@versionNumber>")
+export default new Command("ext:install [extensionName]")
+  .description(
+    "install an extension, provide [extensionName] or [extensionName@versionNumber]; or omit and run with `-i` to see all available extensions."
+  )
   .option("--params <paramsFile>", "name of params variables file with .env format.")
   .before(requirePermissions, ["firebasemods.instances.create"])
   .before(ensureExtensionsApiEnabled)
   .action(async (extensionName: string, options: any) => {
+    const projectId = getProjectId(options, false);
+    const paramFilePath = options.params;
+    let learnMore = false;
+    if (!extensionName) {
+      if (options.interactive) {
+        learnMore = true;
+        extensionName = await promptForOfficialExtension(
+          "Which official extension would you like to install?\n" +
+            "  Select an extension, then press Enter to learn more."
+        );
+      } else {
+        throw new FirebaseError(
+          `Please provide an extension name, or run ${clc.bold(
+            "firebase ext:install -i"
+          )} to see all available extensions`
+        );
+      }
+    }
+
     try {
-      const projectId = getProjectId(options, false);
-      const paramFilePath = options.params;
-      const sourceUrl = await resolveSource(extensionName);
+      const sourceUrl = await resolveSource(
+        extensionName,
+        `Unable to find extension source named ${clc.bold(extensionName)}. ` +
+          `Run ${clc.bold(
+            "firebase ext:install -i"
+          )} to select from the list of all available extensions.`
+      );
       const source = await extensionsApi.getSource(sourceUrl);
+      if (learnMore) {
+        utils.logLabeledBullet(
+          logPrefix,
+          `You selected: ${clc.bold(source.spec.displayName)}.\n` +
+            `${source.spec.description}\n` +
+            `View details: https://firebase.google.com/products/extensions/${extensionName}\n`
+        );
+        const confirm = await promptOnce({
+          type: "confirm",
+          default: true,
+          message: "Would you like to install this extension?",
+        });
+        if (!confirm) {
+          return;
+        }
+      }
```

Comment:
All of this resolving source and printing more probably doesn't need to be in the `try` block that returns an error saying "Error occurred installing the extension". We've not actually gotten to the installing part - can the try block be shrunk a bit?

Reply:
Separated out into 2 try blocks.